src/index.ts

```diff
@@ -0,0 +1,349 @@
+            },
+            {
+                name: "no-box",
+                alias: "Q",
+                description:
+                    "Checker don't use boxes to display results. Only successful domains and line breaks are displayed. \n Use this with {bold --quiet}, it can be easily integrated with other programs.",
+                type: Boolean,
+                defaultValue: false
+            },
+            {
+                name: "add-tld",
+                alias: "t",
+                description:
+                    "Enter additional top-level domains separated by spaces. If you use only the flag, show interactive prompt."
+            },
+            {
+                name: "domain",
+                alias: "d",
+                description:
+                    "{underline TopDomainChecker prioritizes arguments over standard inputs}. \n If it missing, show the interactive prompt. \n Also, {bold --domain} is specified as the default argument and is not required."
+            }
+        ]
+    },
+    {
+        content: `Project home: ${terminalLink(
+            "GitHub",
+            "https://github.com/P2P-Develop/TopDomainChecker"
+        )}`
+    }
+]);
+
+// Arguments cast interface
+type arguments = {
+    version: boolean,
+    verbose: boolean,
+    domain: string[],
+    help: boolean,
+    quiet: boolean,
+    "no-box": boolean,
+    "add-tld": string[],
+    "dry-run": boolean
+}
+
+const arguments_ = commandLineArgs([
+    { name: "version", alias: "V", type: Boolean, defaultValue: false },
+    { name: "verbose", alias: "v", type: Boolean, defaultValue: false },
+    {
+        name: "domain",
+        alias: "d",
+        type: String,
+        multiple: true,
+        defaultOption: true,
+        defaultValue: [""]
+    },
+    { name: "help", alias: "h", type: Boolean, defaultValue: false },
+    { name: "quiet", alias: "q", type: Boolean, defaultValue: false },
+    { name: "no-box", alias: "Q", type: Boolean, defaultValue: false },
+    { name: "add-tld", alias: "t", type: String, defaultValue: [""], multiple: true },
+    { name: "dry-run", alias: "D", type: Boolean, defaultValue: false }
+]) as arguments;
+
+if (arguments_.version) {
+    console.log("v3.1.1");
+    console.log(arguments_);
+    process.exit(0);
+}
+
+if (arguments_.help) {
+    // Show usages if has argument "--help"
+    console.log(usage);
+    process.exit(0);
+}
+
+if (!arguments_["dry-run"] && arguments_.quiet && arguments_.verbose) {
+    // quiet and verbose cannot be used at the same time without --dry-run
+    console.warn(`${chalk.yellowBright.inverse.bold(`  ${figures.warning}  `)} ${chalk.bold("--quiet")} and ${chalk.bold("--verbose")} cannot be used at same time! Replaced with default value!`);
+    [ arguments_.quiet, arguments_.verbose ] = [ false, false ];
+}
+
+const addTld = [ "co.jp", "or,jp", "ne.jp", "ac.jp", "ad.jp", "ed.jp", "go.jp", "gr.jp", "lg.jp" ];
+
+const requestGet = () => {
+    let lockFlag = false;
+
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/P2P-Develop/TopDomainChecker/pullRequest?prid=6364165)